File: 프로그래머스/unrated/181903. qr code/qr code.js
// code.Array.fill().map으로 code의 길이만큼 idx배열 생성
// filter로 q로 나누면 r값이 나오는 idx배열 새로 생성
// for-of문으로 빈 배열 ans에 push
// ans.join(')

function solution(q, r, code) {
    let ans = []

    let length = Array(code.length).fill().map((_, idx) => idx)
    let filter = length.filter(item => item % q == r)

    for (let i of filter) {
        ans.push(code[i])
    }

    return ans.join('')
}

File: 프로그래머스/unrated/181903. qr code/qr code.java
class Solution {
  public String solution(int q, int r, String code) {
    // code를 배열로 만든다.
    String[] str = code.split("");
    
    String result = "";
   // 반복문, 각 인덱스를 q로 나눠 r인 것들을 배열에 담는다.
    for (int i = 0; i < code.length(); i++) {
      if (i % q == r) {
        result += str[i];
      }
    }
   return result;
  }
}